skills/bmad-workflow-builder/scripts/tests/test_prepass_workflow_integrity.py

```
@@ -0,0 +1,221 @@
+"""
+
+
+def test_cross_reference_no_false_positive_critical_for_steps_c(tmp_skill):
+    """The bug this PR fixes: steps-c references must not produce missing-stage criticals."""
+    _write(tmp_skill / 'SKILL.md', _SKILL_MD_STEPS_C)
+    _write(tmp_skill / 'steps-c' / 'step-01-detect.md')
+    _write(tmp_skill / 'steps-c' / 'step-02-write.md')
+
+    summary, findings = _mod.cross_reference_stages(tmp_skill, _SKILL_MD_STEPS_C)
+    missing = [f for f in findings if f.get('category') == 'missing-stage']
+    orphaned = [f for f in findings if f.get('category') == 'naming' and 'orphaned' in f.get('issue', '').lower() or 'not referenced' in f.get('issue', '').lower()]
```

Ruff (0.15.11)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)